crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -539,6 +544,31 @@ where
     Ok(())
 }
 
+#[derive(Clone, Debug)]
+pub struct BlockHeightSubscriptionHandle {
+    rx_block_height: tokio::sync::watch::Receiver<BlockHeight>,
+}
+
+impl BlockHeightSubscriptionHandle {
+    pub async fn wait_for_block_height(
+        &mut self,
+        height: BlockHeight,
+    ) -> anyhow::Result<()> {
```

Comment:
You can return the new block height from `wait_for_block_height`. In this case you don't need `latest_seen_block_height`